lib/iris/tests/unit/common/metadata/test_BaseMetadata.py

```diff
@@ -54,14 +58,76 @@ def test__fields(self):
             "units",
             "attributes",
         )
-        self.assertEqual(BaseMetadata._fields, expected)
+        self.assertEqual(expected, self.cls._fields)
+
+
+class Test___eq__(tests.IrisTest):
+    def setUp(self):
+        self.kwargs = dict(
+            standard_name=sentinel.standard_name,
+            long_name=sentinel.long_name,
+            var_name=sentinel.var_name,
+            units=sentinel.units,
+            attributes=sentinel.attributes,
+        )
+        self.cls = BaseMetadata
+        self.metadata = self.cls(**self.kwargs)
+
+    def test_lenient_service(self):
+        qualname___eq__ = _qualname(self.cls.__eq__)
+        self.assertIn(qualname___eq__, _LENIENT)
+        self.assertTrue(_LENIENT[qualname___eq__])
+        self.assertTrue(_LENIENT[self.cls.__eq__])
+
+    def test_cannot_compare_non_class(self):
+        result = self.metadata.__eq__(None)
+        self.assertIs(NotImplemented, result)
+
+    def test_cannot_compare(self):
+        other = CubeMetadata(*(None,) * len(CubeMetadata._fields))
+        result = self.metadata.__eq__(other)
+        self.assertIs(NotImplemented, result)
```

Comment:
I would prefer a more descriptive name for this test. Maybe something lik
`test_cannot_compare_diff_class`